postgres/datadog_checks/postgres/statements.py

```diff
@@ -96,21 +98,25 @@ def _get_pg_stat_statements_columns(self, db):
         be upgraded without upgrading extensions, even when the extension is included by default.
         """
         # Querying over '*' with limit 0 allows fetching only the column names from the cursor without data
+        if self._stat_column_cache:
```

Comment:
If I get that right, this cached is never refreshed. The StatementMetrics instance is initialized just once https://github.com/DataDog/integrations-core/blob/4dcb5e6acd2c3607e694f2d45114ef1b07880496/postgres/datadog_checks/postgres/postgres.py#L44
on the first check run. And the `self._stat_column_cache` is only populated the first time _get_pg_stat_statements_columns is called.

This means that in case the extension or the db gets updated, or if the endpoint used by the check somehow points to a different database, this variable will always be out of date and lead to failures down the line.

A good approach to solve this would be to:
1. Identify where the code would fail if `self._stat_column_cache` has the wrong list of column names.
2. Catch the error and refresh that variable.
3. Potentially retry the operation that raised the error if a 15 seconds gap cannot be accepted.

Reply:
Great find @FlorianVeaux! Pushed up the cache reset change. As you pointed out, if we do come across this scenario it will now be able to heal itself by resetting the cache upon an error and querying for the correct columns next time it runs.